File: HumanComputerInteraction/HelloAndroid/Droid/MainActivity.cs
using Android.App;
using Android.Widget;
using Android.OS;

namespace HelloAndroid.Droid
{
    [Activity(Label = "HelloAndroid", MainLauncher = true, Icon = "@mipmap/icon")]
    public class MainActivity : Activity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.Main);

            // Get our button from the layout resource,
            // and attach an event to it
            Button button = FindViewById<Button>(Resource.Id.myButton);
            EditText editText = FindViewById<EditText>(Resource.Id.editText1);
            TextView textView = FindViewById<TextView>(Resource.Id.helloText);

            button.Click += delegate {

                string name = editText.Text;

                textView.Text = "Hello " + (string.IsNullOrWhiteSpace(name) ? "Android" : name) + "!";

            };
        }
    }
}

File: HumanComputerInteraction/HelloWorld/app/src/main/java/com/stiancronje/helloworld/MainActivity.java
package com.stiancronje.helloworld;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    TextView textView;
    EditText editText;
    Button button;


    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView1);
        editText = findViewById(R.id.editText1);
        button = findViewById(R.id.button1);

        button.setOnClickListener(new ButtonListener());
    }

    private class ButtonListener implements View.OnClickListener
    {
        public void onClick(View V)
        {
            String name = editText.getText().toString();

            if(!name.equals(""))
            {
                textView.setText(String.format("%s %s!", getString(R.string.hello) ,name));
            }
            else
            {
                textView.setText(getString(R.string.enter_name));
            }
        }
    }

}
